src/java/org/apache/cassandra/journal/Flusher.java

```diff
@@ -97,20 +102,144 @@ void start()
     void shutdown()
     {
         flushExecutor.shutdown();
+        if (fsyncExecutor != null)
+            fsyncExecutor.shutdown();
     }
 
     @Simulate(with={MONITORS,GLOBAL_CLOCK,LOCK_SUPPORT})
     private class FlushRunnable implements Interruptible.Task
     {
-        private final MonotonicClock clock;
+        @Simulate(with={MONITORS,GLOBAL_CLOCK,LOCK_SUPPORT})
+        private class FSyncRunnable implements Interruptible.Task
+        {
+            // this is written only by the Flusher thread, and read only by the Fsync thread
+            ActiveSegment<K, V> fsyncTo;
```

Comment:
nit: how about `fsyncUpTo`? Not a native speaker, but when I see the variable first association was that this is a target of fsync.